Mathlib/SetTheory/Ordinal/Basic.lean

```diff
@@ -1213,6 +1216,42 @@ theorem enum_zero_eq_bot {o : Ordinal} (ho : 0 < o) :
   rfl
 #align ordinal.enum_zero_eq_bot Ordinal.enum_zero_eq_bot
 
+instance {o : Ordinal} : IsWellOrder (Iio o) (· < ·) := IsWellOrder.mk
+
+theorem type_Iio (o : Ordinal.{u}) : type (α := Iio o) (· < ·) = lift.{u + 1, u} o := by
+  let f : o.out.α → Iio o := fun x ↦
+    ⟨typein ((· < ·) : o.out.α → o.out.α → Prop) x, typein_lt_self x⟩
+  have fsur : Function.Surjective f := fun b ↦
+    ⟨enum (· < ·) b.val
+        (by
+          rw [type_lt]
+          exact b.prop),
+      Subtype.ext (typein_enum _ _)⟩
+  have finj : Function.Injective f := fun _ _ _ ↦ by
+    simp_all only [Subtype.mk.injEq, typein_inj, f]
+  have : ((· < ·) : o.out.α → o.out.α → Prop) ≃r ((· < ·) : Iio o → Iio o → Prop) := by
+    use Equiv.ofBijective f ⟨finj, fsur⟩
+    intro a b
+    simp_all only [Equiv.ofBijective_apply, Subtype.mk_lt_mk, typein_lt_typein, f]
+  have : ((· < ·) : (ULift.{u + 1, u} o.out.α) → (ULift.{u + 1, u} o.out.α) → Prop) ≃r
+      ((· < ·) : Iio o → Iio o → Prop) := by
+    refine' ⟨⟨fun ⟨x⟩ ↦ this x, fun x ↦ ⟨this.invFun x⟩, _, this.1.4⟩, this.map_rel_iff'⟩
+    · convert this.1.3
+      exact (iff_true_right this.1.3).mpr fun x ↦ congrArg ULift.up (this.1.3 x.down)
+  have aux : IsWellOrder (ULift.{u + 1, u} (Quotient.out o).α) (· < ·) := by
+    have : o.out.r ≃r fun x y : ULift.{u + 1, u} o.out.α ↦ x < y :=
+      ⟨⟨fun x ↦ ⟨x⟩, by aesop, congrFun rfl, congrFun rfl⟩, by aesop⟩
+    exact this.isWellOrder
+  convert this.symm.ordinal_type_eq
+  have : type fun x x_1 : ULift.{u + 1, u} (Quotient.out o).α ↦ x < x_1 =
+      lift.{u + 1, u} (type fun x x' : (Quotient.out o).α ↦ x < x') := rfl
+  rw [this, type_lt]
+
+theorem relIso_nat_omega : Nonempty (((· < ·) : ℕ → ℕ → Prop)
```

Comment:
We could give an explicit isomorphism instead. This would be placed near the end of `SetTheory.Ordinal.Arithmetic`.
```lean 4
def relIso_nat_omega : ℕ ≃o Iio ω where
  toFun n := ⟨n, nat_lt_omega n⟩
  invFun n := Classical.choose (lt_omega.1 n.2)
  left_inv n := by
    have h : ∃ m : ℕ, n = (m : Ordinal) := ⟨n, rfl⟩
    exact (Ordinal.natCast_inj.1 (Classical.choose_spec h)).symm
  right_inv n := Subtype.eq (Classical.choose_spec (lt_omega.1 n.2)).symm
  map_rel_iff' := @natCast_le
  ```